src/router.rs

```diff
@@ -24,6 +25,17 @@ pub struct HttpRoutes<T> {
     routes: HashMap<String, Box<dyn EndpointHandler<T> + Sync + Send>>,
 }
 
+impl<T> fmt::Debug for HttpRoutes<T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("HttpRoutes")
+            .field("server_id", &self.server_id)
+            .field("prefix", &self.prefix)
+            .field("media_type", &self.media_type)
+            .field("routes", &"{ .. }")
```

Comment:
changing this to : ```.field("routes", &self.routes)```

and implementing Debug for EndpointHandler which just has :
```
impl std::fmt::Debug for dyn EndpointHandler{
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        f.debug_struct("EndpointHandler")
        .field("handle_request",&"{..}}")
        .finish()
    }
}
```
can improve the debug prints to give us the routes strings

Reply:
`impl std::fmt::Debug for dyn EndpointHandler` cannot be done.

Reply:
Could you provide the full code for this change, I am not convinced this is a good solution.

Reply:
This is what I tried: 
```
diff --git a/src/router.rs b/src/router.rs
index 4676459..cc80bf9 100644
--- a/src/router.rs
+++ b/src/router.rs
@@ -16,6 +16,14 @@ pub trait EndpointHandler<T>: Sync + Send {
     fn handle_request(&self, req: &Request, arg: &T) -> Response;
 }
 
+impl<T> fmt::Debug for dyn EndpointHandler<T> + Send + Sync + 'static{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("EndpointHandler")
+            .field("handle_request", &"{ .. }")
+            .finish()
+    }
+}
+
 /// An HTTP routes structure.
 pub struct HttpRoutes<T> {
     server_id: String,
@@ -31,7 +39,7 @@ impl<T> fmt::Debug for HttpRoutes<T> {
             .field("server_id", &self.server_id)
             .field("prefix", &self.prefix)
             .field("media_type", &self.media_type)
-            .field("routes", &"{ .. }")
+            .field("routes", &self.routes)
             .finish()
     }
 }
```

and the output for this shows:
```
HttpRoutes {
    server_id: "Mock_Server",
    prefix: "/api/v1",
    media_type: ApplicationJson,
    routes: {
        "GET:/api/v1/func1": EndpointHandler {
            handle_request: "{ .. }",
        },
    },
}
```
vs below with existing code in PR
```
HttpRoutes {
    server_id: "Mock_Server",
    prefix: "/api/v1",
    media_type: ApplicationJson,
    routes: "{ .. }",
}
```